auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BasicAuthToBearerTokenFilter.java

```diff
@@ -0,0 +1,122 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tokens.auth.http;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.io.BaseEncoding;
+import com.google.common.net.HttpHeaders;
+import com.palantir.tokens.auth.AuthHeader;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link Filter} that replaces basic auth with a bearer token.
+ * <p>
+ * It assumes that the bearer token is held in the password field of the basic auth credentials. It assumes that the
+ * bearer token is base-64 encoded.
+ */
+public class BasicAuthToBearerTokenFilter implements Filter {
+    private static final Logger log = LoggerFactory.getLogger(BasicAuthToBearerTokenFilter.class);
+
+    private static final String BASIC_AUTH_STR = "Basic";
+    private static final BaseEncoding BASE_64_ENCODING = BaseEncoding.base64Url();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public final void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        ServletRequest updatedRequest = addBearerTokenIfBasicAuth(request);
+        chain.doFilter(updatedRequest, response);
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    private ServletRequest addBearerTokenIfBasicAuth(ServletRequest request) {
+        if (request instanceof HttpServletRequest) {
+            HttpServletRequest httpRequest = (HttpServletRequest) request;
+            String rawAuthHeader = httpRequest.getHeader(HttpHeaders.AUTHORIZATION);
+            return addBearerTokenIfBasicAuth(httpRequest, rawAuthHeader);
+        } else {
+            log.warn("Request is not an HttpServletRequest.");
+            return request;
+        }
+    }
+
+    private ServletRequest addBearerTokenIfBasicAuth(HttpServletRequest request, String rawAuthHeader) {
+        if (isBasicAuth(rawAuthHeader)) {
+            AuthHeader authHeader;
+            try {
+                authHeader = base64DecodePassword(rawAuthHeader);
+            } catch (IllegalArgumentException e) {
+                log.warn("IllegalArgumentException: " + e.getMessage());
```

Comment:
probably want to say "Could not decode password in basic auth header". Also, since you have the exception, can just log the actual exception:

    log.warn("Could not decode password in basic auth header", e);